ReactNativeClient/lib/services/searchengine/queryBuilder.ts

```diff
@@ -0,0 +1,361 @@
+const { time } = require('lib/time-utils.js');
+
+const tagFilter = (tags: string[], intersect: boolean = true, negated: boolean = false) => { // note_tags.tag_id  (NOT) IN (SELECT
+	let result = new Array(tags.length).fill(`${intersect ? 'INTERSECT' : 'UNION'} SELECT note_tags.note_id FROM note_tags WHERE note_tags.tag_id IN (SELECT tags.id from tags WHERE tags.title LIKE ?)`).join(' ');
```

Comment:
Could you put the long queries in this file over multiple lines, so that it's easier to read? eg

```sql
SELECT note_tags.note_id
FROM note_tags
WHERE note_tags.tag_id IN (
    SELECT tags.id from tags
    WHERE tags.title
    LIKE ?
)
```